src/Aspire.Hosting/ApplicationModel/CommandLineArgsCallbackAnnotation.cs

```diff
@@ -50,6 +50,17 @@ public CommandLineArgsCallbackAnnotation(Action<IList<object>> callback)
 /// <param name="cancellationToken"> The cancellation token associated with this execution.</param>
 public sealed class CommandLineArgsCallbackContext(IList<object> args, CancellationToken cancellationToken = default)
 {
+    private readonly IResource? _resource;
+
+    /// <summary>
+    /// Represents a callback context for the list of command-line arguments associated with an executable resource.
+    /// </summary>
+    /// <param name="args"> The list of command-line arguments.</param>
+    /// <param name="resource"> The resource associated with this callback context.</param>
+    /// <param name="cancellationToken"> The cancellation token associated with this execution.</param>
+    public CommandLineArgsCallbackContext(IList<object> args, IResource? resource = null, CancellationToken cancellationToken = default)
```

Comment:
Now that we are introducing a non-primary constructor for consistency and readability we should move the primary constructor into a regular constructor as well.

Reply:
Back compat

Reply:
Wait you're right @mitchdenny  ths shouldn't be nullable.